integration-tests/src/user/runtime_user.rs

```diff
@@ -80,6 +80,7 @@ impl RuntimeUser {
         apply_state: ApplyState,
         prev_receipts: Vec<Receipt>,
         transactions: Vec<SignedTransaction>,
+        use_flat_storage: bool,
```

Comment:
Hm, I don't understand this flag, why wouldn't we want to use flat state in all tests?

Reply:
This is needed to check fallback to Trie, to defend from accidental protocol changes - if at some point we decide to implement this fallback, the new test requires doing it properly.
And such fallback is needed for view clients anyway, so I think it is valuable to have it here.